charts/redpanda/console.tpl.go

```diff
@@ -20,10 +20,192 @@ import (
 	"fmt"
 
 	"github.com/redpanda-data/console/backend/pkg/config"
+	"github.com/redpanda-data/helm-charts/charts/console"
 	"github.com/redpanda-data/helm-charts/pkg/gotohelm/helmette"
+	"github.com/redpanda-data/helm-charts/pkg/kube"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/utils/ptr"
 )
 
-func ConsoleConfig(dot *helmette.Dot) any {
+// To connect Console with Redpanda the values are adjusted.
+func renderConsole(dot *helmette.Dot) []kube.Object {
+	v := helmette.UnmarshalInto[Values](dot.Values)
+
+	if !ptr.Deref(v.Console.Enabled, true) {
+		return nil
+	}
+
+	consoleDot := dot.Subcharts["console"]
+
+	consoleValue := helmette.UnmarshalInto[console.Values](consoleDot.Values)
+	if !ptr.Deref(v.Console.Secret.Create, false) {
+		consoleValue.Secret.Create = true
+		if license := GetLicenseLiteral(dot); license != "" {
+			consoleValue.Secret.Enterprise = console.EnterpriseSecrets{License: ptr.To(license)}
+		}
+	}
+
+	if !ptr.Deref(v.Console.ConfigMap.Create, false) {
+		consoleValue.ConfigMap.Create = true
+		consoleValue.Console.Config = ConsoleConfig(dot)
+	}
+
+	if !ptr.Deref(v.Console.Deployment.Create, false) {
+		consoleValue.Deployment.Create = true
+		extraVolumes := []corev1.Volume{}
+		extraVolumeMounts := []corev1.VolumeMount{}
+		extraEnvVars := []corev1.EnvVar{}
+		if v.Auth.IsSASLEnabled() {
+			command := append([]string{},
+				"sh",
+				"-c",
+				"set -e; IFS=':' read -r KAFKA_SASL_USERNAME KAFKA_SASL_PASSWORD KAFKA_SASL_MECHANISM < <(grep \"\" $(find /mnt/users/* -print));",
+				fmt.Sprintf(" KAFKA_SASL_MECHANISM=${KAFKA_SASL_MECHANISM:-%s};", SASLMechanism(dot)),
+				" export KAFKA_SASL_USERNAME KAFKA_SASL_PASSWORD KAFKA_SASL_MECHANISM;",
+				" export KAFKA_SCHEMAREGISTRY_USERNAME=$KAFKA_SASL_USERNAME;",
+				" export KAFKA_SCHEMAREGISTRY_PASSWORD=$KAFKA_SASL_PASSWORD;",
+				" export REDPANDA_ADMINAPI_USERNAME=$KAFKA_SASL_USERNAME;",
+				" export REDPANDA_ADMINAPI_PASSWORD=$KAFKA_SASL_PASSWORD;",
+				" /app/console $@",
+				" --")
+			consoleValue.Deployment.Command = command
+			extraVolumes = append(extraVolumes, corev1.Volume{
+				Name: fmt.Sprintf("%s-users", Fullname(dot)),
+				VolumeSource: corev1.VolumeSource{
+					Secret: &corev1.SecretVolumeSource{
+						SecretName: v.Auth.SASL.SecretRef,
+					},
+				},
+			})
+			extraVolumeMounts = append(extraVolumeMounts, corev1.VolumeMount{
+				Name:      fmt.Sprintf("%s-users", Fullname(dot)),
+				MountPath: "/mnt/users",
+				ReadOnly:  true,
+			})
+		}
+
+		if v.Listeners.Kafka.TLS.IsEnabled(&v.TLS) {
+			certName := v.Listeners.Kafka.TLS.Cert
+			cert := v.TLS.Certs.MustGet(certName)
+			secretName := fmt.Sprintf("%s-%s-cert", Fullname(dot), certName)
+			if cert.SecretRef != nil {
+				secretName = cert.SecretRef.Name
+			}
+			if cert.CAEnabled {
```

Comment:
You can either solve this here or file a ticket handle it later: truststores (Defined on the listener instead of the cert) need to be handled here as they can also be used to set the CA.

Reply:
I think the Volumes and VolumeMounts have now much more clear functions.